client/src/main/java/com/linecorp/decaton/client/internal/DecatonTaskProducer.java

```diff
@@ -63,7 +63,15 @@ public DecatonTaskProducer(String topic, Properties producerConfig,
 
     public CompletableFuture<PutTaskResult> sendRequest(byte[] key, DecatonTaskRequest request) {
         ProducerRecord<byte[], DecatonTaskRequest> record = new ProducerRecord<>(topic, key, request);
+        return sendRequest(record);
+    }
+
+    public CompletableFuture<PutTaskResult> sendRequest(byte[] key, DecatonTaskRequest request, int partition) {
```

Comment:
DecatonTaskProducer is in `internal` package so not considered to be public API, so we don't need to keep backward compatibility. Let's just change original `sendRequest` to receive `Integer partition`